docs/data_structures/sim_specs.rst

```diff
@@ -1,44 +1,66 @@
 .. _datastruct-sim-specs:
 
-sim_specs
-=========
-Used to specify the simulation function, its inputs and outputs, and user data::
+Simulation Specs
+================
 
-    sim_specs: [dict]:
+Used to specify the simulation function, its inputs and outputs, and user data.
 
-        'sim_f' [func]:
-            the simulation function being evaluated
-        'in' [list]:
-            field names (as strings) to be given to sim_f by alloc_f
-        'persis_in' [list]:
-            field names (as strings) that will be given back to a persistent sim_f
-        'out' [list of tuples (field name, data type, [size])]:
-            sim_f outputs to be stored in the libEnsemble history
-        'user' [dict, optional]:
-            Data structure to contain problem specific constants and/or input data
+Can be constructed and passed to libEnsemble as a Python class or a dictionary. When provided as a Python class,
+all data is validated immediately on instantiation. When provided as a dictionary, all data is validated
+upon passing into :meth:`libE()<libensemble.libE.libE>`.
 
-.. note::
-  * The entirety of ``sim_specs`` is passed from the worker each time a
-    simulation is requested by the allocation function.
+As a Python class:
 
-  * The tuples in ``sim_specs['out']`` are entered into the manager's
-    :ref:`history array<datastruct-history-array>`.
+.. autopydantic_model:: libensemble.specs.SimSpecs
+  :model-show-json: False
+  :members:
 
-.. seealso::
+.. code-block:: python
+    :linenos:
 
-  .. _sim-specs-example1:
+    ...
+    import numpy as np
+    from libensemble.specs import SimSpecs
+    from simulator import sim_find_sine
+    ...
 
-  - test_uniform_sampling.py_ has a ``sim_specs``  that declares
-    the name of the ``'in'`` field variable, ``'x'`` (as specified by the
-    corresponding generator ``'out'`` field ``'x'`` from the :ref:`gen_specs
+    sim_specs = SimSpecs(
+      sim_f = sim_find_sine,
+      inputs = ["x"],
+      out = [("y", float)]
+    )
+    ...
+
+As a dictionary:
+
+.. code-block:: python
+    :linenos:
+
+    ...
+    import numpy as np
+    from simulator import six_hump_camel
+    ...
+
+    sim_specs = {
+      "sim_f": six_hump_camel,
+      "in": ["x"],
+      "out": [("y", float)],
+    }
+    ...
+
+.. _sim-specs-example1:
+
+  - test_uniform_sampling.py_ has a :class:`sim_specs<libensemble.specs.SimSpecs>`  that declares
```

Comment:
broken link - this should really be a regression test?

Reply:
@shuds13 Is there an automated way to test that links aren't broken?